vtm.js

```diff
@@ -0,0 +1,339 @@
+          tmp[this.stats.data[i].list[j].value]++;
+        }
+      }
+    tmp.forEach((element, index) => tmp[index] = this.stats.resource[index] - element);
+    this.resourceCount = tmp;
+  }
+})
+
+//STAT CATEGORY
+app.component('stat-category',{
+  props: ['categ','resource','scale','parent'],
+  template:`<div class="statList"><h2>{{categ.display}}</h2>
+             <ul class="ulStats">
+             <li v-for="item in categ.list">
+             <stat 
+               :stat="item"
+               :resource="resource"
+               :scale="scale"
+               :parent="parent"></stat></li></ul></div>`
+})
+
+
+//STAT
+app.component('stat',{
+  props: ['stat', 'resource', 'scale', 'parent'],
+  data() {
+    return {
+      initialValue : this.stat.value,
+      hoverPointer : null,
+      hoverToggle : false
+    }},
+  template: `<div>  
+    <div :class="{stat : true, mOver : hoverToggle}">        
+      <div class="statName">{{stat.display}}</div>
+      <div v-if="stat.specialty" class="specialty">{{stat.specialty}}</div>
+      <div class="points">  
+        <span v-for="i in scale"       
+          :class="pointClass(i)"
+          @click = "handleClick()"
+          @mouseover = "handleHover(i , true)"
+          @mouseleave = "handleHover(i , false)"        
+          :value = i
+        ></span>          
+      </div>       
+      </div>
+    </div>`,
+        
+    methods: {
+      pointClass(i){
+        const init_val = this.initialValue;
+        const ptr = this.hoverPointer;
+        const val = this.stat.value;
+        return {
+             point : true,
+             init : i<=init_val,
+             fill : i>init_val &&i<=val,
+             mOver : (ptr && (i === ptr || ((i>ptr) !== (i>val))))
+             }
+      },
+      handleHover(i, hover){
+        if (!hover) i=null;
+        this.hoverToggle = hover;
+        this.hoverPointer = i; 
+      },
+      handleClick(){
+        //debugger;
+        const init = this.initialValue;
+        var value = this.stat.value;
+        var i = this.hoverPointer;
+        var diff = i - value;
+        let tmp;
+        
+        // -1 if not found || i if found [0, index]
+        const addCheck = (array, index) =>{
+          var i;
+          for (i = index; i >=0 ;i--){
+            if (array[i]>0) break;
+          }          
+          return i;
+        }
+        //adding points : check limit
+        if (diff > 0) {
+          tmp = addCheck(this.resource,i);
+          //if (tmp <= 0) i = value;
+          //else i = tmp;
+          i = (tmp > value) ? tmp : value;
+        }
+        //clicked filled point -> removing points, not below limit aka initial value
+        if ((diff <= 0)&&(i>init)) i--;             
+    
+        //OK setting values 
+        //emit: change value count
+        this.resource[value]+=1;
+        this.resource[i]+=-1;
+        this.stat.value = i;
+      }
+    }     
+})
+/*
+classNamePoint(i){
```

Comment:
dalsi comment ktory sa moze dat prec

Reply:
jo ale je tam ta myslienka s tym bonusom z itemov ktora neni este domyslena, v podstate je to tam len preto aby som na to nezabudol a lebo som prilis lenivy aby som tu myslienku uspornejsie zformuloval